content-src/components/LinkMenu/LinkMenu.js

```diff
@@ -38,9 +38,16 @@ const LinkMenu = React.createClass({
           label: "Dismiss",
           icon: "dismiss",
           userEvent: "BLOCK",
-          onClick: () => dispatch(actions.NotifyBlockURL(site.url))
+          onClick: () => {
+            if (site.recommended) {
+              dispatch(actions.NotifyBlockRecommendation(site.url));
+              dispatch(actions.RequestHighlightsLinks());
```

Comment:
If you turned it into a `RequestExpect` instead of a `Notify` you could do it, but it will require some refactoring. This is kinda a problem with all the delete/block stuff right now, let's leave this the way it is for now and file a follow-up to address the side effects stuff separately